pkg/apiserver/statement/queries.go

```diff
@@ -17,15 +17,93 @@ import (
 	"fmt"
 	"regexp"
 	"sort"
+	"strconv"
 	"strings"
 
 	"github.com/jinzhu/gorm"
 )
 
 const (
-	statementsTable = "INFORMATION_SCHEMA.CLUSTER_STATEMENTS_SUMMARY_HISTORY"
+	statementsTable        = "INFORMATION_SCHEMA.CLUSTER_STATEMENTS_SUMMARY_HISTORY"
+	stmtEnableVar          = "tidb_enable_stmt_summary"
+	stmtRefreshIntervalVar = "tidb_stmt_summary_refresh_interval"
+	stmtHistroySizeVar     = "tidb_stmt_summary_history_size"
 )
 
+type sqlVariable struct {
+	Name  string `gorm:"column:Variable_name"`
+	Value string `gorm:"column:Value"`
+}
+
+// How to get sql variables by GORM
+// https://github.com/jinzhu/gorm/issues/2616
+func querySQLIntVariable(db *gorm.DB, name string) (int, error) {
+	var variables []sqlVariable
+	err := db.Raw("SHOW GLOBAL VARIABLES LIKE ?", name).Scan(&variables).Error
+	if err != nil {
+		return 0, err
+	}
+	strVal := variables[0].Value
+	if strVal == "" {
+		return -1, nil
+	}
+	intVal, err := strconv.Atoi(strVal)
+	if err != nil {
+		return 0, err
+	}
+	return intVal, nil
+}
+
+func QueryStmtConfig(db *gorm.DB) (*Config, error) {
+	config := Config{}
+
+	enable, err := querySQLIntVariable(db, stmtEnableVar)
+	if err != nil {
+		return nil, err
+	}
+	config.Enable = enable != 0
+
+	refreshInterval, err := querySQLIntVariable(db, stmtRefreshIntervalVar)
+	if err != nil {
+		return nil, err
+	}
+	if refreshInterval == -1 {
+		config.RefreshInterval = 1800
+	} else {
+		config.RefreshInterval = refreshInterval
+	}
+
+	historySize, err := querySQLIntVariable(db, stmtHistroySizeVar)
+	if err != nil {
+		return nil, err
+	}
+	if historySize == -1 {
+		config.HistorySize = 24
+	} else {
+		config.HistorySize = historySize
+	}
+
+	return &config, err
+}
+
+func UpdateStmtConfig(db *gorm.DB, config *Config) (err error) {
+	var sql string
+	sql = fmt.Sprintf("SET GLOBAL %s = ?", stmtEnableVar)
+	err = db.Exec(sql, config.Enable).Error
```

Comment:
Tried `db.Exec("SET GLOBAL ? = ?", stmtEnableVar, config.Enable).Error` doesn't work, so the `?` only can be show in value position?

Reply:
yes, I think this is expected.